app/assets/javascripts/discourse/controllers/composer.js.es6

```diff
@@ -775,12 +775,47 @@ export default Ember.Controller.extend({
           .then(resolve, reject);
       }
 
-      // we need a draft sequence for the composer to work
-      if (opts.draftSequence === undefined) {
+      // check if there is another draft saved on server
+      // or get a draft sequence number
+      if (!opts.draft || opts.draftSequence === undefined) {
         return Draft.get(opts.draftKey)
-          .then(function(data) {
-            opts.draftSequence = data.draft_sequence;
-            opts.draft = data.draft;
+          .then(data => {
+            if (!data.draft) {
+              return data;
+            }
+
+            // do not show abandon dialog if old draft is clean
+            const draft = JSON.parse(data.draft);
+            if (draft.reply === draft.originalText) {
+              data.draft = null;
+              return data;
+            }
+
+            return new Promise(resv => {
```

Comment:
Could we extract this in its own function with an explicit name, like `return this.confirmAbandon()` ?

Reply:
Well, it does more than just look for an existent draft and the name of "confirmAbandon" would be misleading (it also gets a draft sequence number in case it is missing).

Initially, it was a different function, but @ZogStriP suggested that I could combine that code with the already existing call to `Draft.get`.

Reply:
Totally feel free to find a better name, that was just a suggestion based on my limited understanding of the code.